shell/browser/api/electron_api_web_contents.cc

```diff
@@ -421,8 +424,11 @@ bool IsDeviceNameValid(const base::string16& device_name) {
 }
 
 base::string16 GetDefaultPrinterAsync() {
-  base::ScopedBlockingCall scoped_blocking_call(FROM_HERE,
-                                                base::BlockingType::MAY_BLOCK);
+#if defined(OS_WIN)
+  // Blocking is needed here because Windows printer drivers are oftentimes
+  // not thread-safe and have to be accessed on the UI thread.
+  PrintingScopedAllowBlocking allow_blocking;
```

Comment:
Can we instead use `base::ScopedAllowBlocking` instead of inheriting testing class ?

Reply:
@deepak1556 no - see comment at top; we would need to patch the class in Chromium to declare this class a friend class

Reply:
Missed that, in that case you can use `base::ThreadRestrictions::ScopedAllowIO` which offers same result and we have been using it in our code.